### ISSUE TITLE
Anticipated Engagements Count

### ISSUE THREAD

**Issue by sethmcknight** [SeedCompany/cord-field#1121]
Often when a Project is in Early Conversations and Concept Development, we have an idea of how many Engagements might be a part of the Project, but don't know the exact Languages, People, or Partners who will be engaged.

Without this number, Leadership is blind to what may be coming in future years, making it difficult to forecast financial and personnel needs. We need a way to capture the number of anticipated engagements in order to provide these projections.

**Comment by elisabeth-sorrell** [comment]
@sethmcknight @sherijennings While discussing this with @AmyRickwartz, it looked like there is a little more to this task than just adding a data button for an engagement count on the Project page. Is the goal for this ticket still the same?

**Comment by sethmcknight** [comment]
<img width="706" alt="Screen Shot 2023-05-09 at 2 32 33 PM" src="https://github.com/SeedCompany/cord-field/assets/7377802/edba3b3a-82d3-4ac2-86a8-6bcba2ba2c58">

**Comment by elisabeth-sorrell** [comment]
@sethmcknight We currently have business rules in place where an engagement can only be added when the project is `In Development` status or are part of a Changeset and `Active` Status. Is it the case that only Unknown engagements can be added when the project is in `Early Conversations`?

**Comment by elisabeth-sorrell** [comment]
@sethmcknight 
Should the project and/or Engagement be in a certain status/step before a Language or Intern can be attached to it?

Should there be any verification before a project or engagement transitions to a different step whether the current engagements are unknown?

**Comment by sethmcknight** [comment]
@elisabeth-sorrell that's correct. This should replace the current Add Engagement functionality and follow the same status/permission rules.

We should check that there are no missing Languages/People when `In Development` and transitioning from `Prep for Consultant Endorsement` or any time an engagement is added after that.
